platform-sdk/docs/components/componentFramework.md

```diff
@@ -0,0 +1,218 @@
+
+Connect the business logic to the TaskScheduler by binding the InputWire to the method that operates on the input data
+and produces the output data. The TaskScheduler is now connected, and will automatically forward the data produced by
+the bound method to the OutputWire. The component is now built.
+
+```
+fooInputWire.bind(logic::intToString);
+```
+
+![Step 4](step4.drawio.svg)
+
+### Step 5 - Connecting Components
+
+Connect the component to other components by soldering input and output wires.
+
+```
+fooTaskScheduler.getOutputWire().solderTo(barInputWire);
+barTaskScheduler.getOutputWire().solderTo(bazInputWire);
+```
+
+![Step 5](step5.drawio.svg)
+
+## Features
+
+### Wiring Model
+
+The wiring model tracks all task schedulers and wires that connect them and is capable of doing useful analysis of the
+resulting graph. Most notably, it can produce mermaid style diagrams that show the flow of data through wires and
+components, detect cyclic backpressure that could result in a deadlock, and verify proper wiring of various task
+schedulers according to their restrictions (for example, a concurrent task scheduler is not permitted to be wired to a
+direct scheduler).
+
+### Wire Add Ons
+
+Wire add ons (or transformers) are simple units of logic that transform the data type on a wire to a different type. For
+example, a `WireListSplitter` takes data from a wire with a `List` type and streams the contents of the list, one by
+one, onto another wire. Wire transformers operate on the thread defined in the task scheduler and do not come with their
+own concurrency logic. Some types of transformers are described below:
+
+**Wire Filter:** A predicate that operates on the data type of the output wire. Data is only passed to the filter output
+wire if the
+predicate passes.
+
+**Wire Transformer:** A function that operates on the data type of the output wire. Data produced by the wire
+transformer and forwarded on its
+output wire may be a different type than the input.
+
+**Wire List Splitter:** A function that operates on a list of data and streams the individual items on the output wire.
+
+### Backpressure
+
+Backpressure can be configured for some task schedulers. If available and configured, threads are blocked from
+adding data to a component that has reached the limit of unprocessed tasks. This feature prevents too many tasks from
+building up in a given component and signals the components feeding into it to back off so that it can catch up. In
+order to prevent circular data flows that could result in a deadlock of backpressure, task schedulers provide a way to
+bypass the backpressure mechanism. The method `inject()` ignores the number of unprocessed tasks and the maximum
+configured value in order to prevent such deadlocks. **If backpressure is desired, the primary path of data flow must
+always pass data using the backpressure methods. Secondary data flows must use `inject()` to avoid deadlock**. Circular
```

Comment:
> **If backpressure is desired, the primary path of data flow must
always pass data using the backpressure methods. Secondary data flows must use `inject()` to avoid deadlock**. 

In the Wire section, the document explains that "Wires are the construct on which data flows between components" 
Now, it introduces the "path of data flows" concept and contrasts a primary data flow with a secondary data flow.
I cannot help wondering how a path of data flow is formed. Let's say I have a multiple input wire/output wire component, What defines a primary path from a secondary one? Are secondary Output wires involved?

Reply:
I can understand now how the double use of "primary" and "secondary" could be confusing. The primary data flow path mentioned here in the context of back pressure is conceptual and not directly related to primary or secondary output wires. It is the flow of data on which back pressure is required to slow down input to the system. I think it makes sense to use the primary output wires for this flow of data, but it is not required by the framework.

Reply:
updated to:

```
**If backpressure is desired, only a single non-cyclic data flow
should use backpressure. Any data flow that forms a cycle should use `inject()` to avoid deadlock.**
```